kola/tests/misc/gcp.go

```diff
@@ -0,0 +1,27 @@
+// Copyright The Mantle Authors.
+// SPDX-License-Identifier: Apache-2.0
+package misc
+
+import (
+	"github.com/flatcar-linux/mantle/kola/cluster"
+	"github.com/flatcar-linux/mantle/kola/register"
+)
+
+func init() {
+	register.Register(&register.Test{
+		Name:        "cl.misc.gce.oem",
+		ClusterSize: 1,
+		Platforms:   []string{"gce"},
+		Distros:     []string{"cl"},
+		Run:         gceVerifyOEMService,
+	})
+}
+
+func gceVerifyOEMService(c cluster.TestCluster) {
+	// verify that the oem-gce service is running
+	c.MustSSH(c.Machines()[0], "systemctl is-active oem-gce.service")
+	nrestarts := c.MustSSH(c.Machines()[0], "systemctl show oem-gce.service -P NRestarts")
+	if string(nrestarts) != "0" {
```

Comment:
With the sysext workaround I had created last week it could lead to a restart, but I think we can add `After=…` to the oem-gce service to prevent a restart.

Also, maybe first do `systemctl is-system-running --wait` to prevent races?

Reply:
`platform.CheckMachine ` is run in `platform.StartMachine` and loops on `systemctl is-system-running`, so this is already covered. The oem-service doesn't show up as failed because it's a "restart always" service, I think that's why we didn't catch this in the first place.

I think the oem service should not restart because `systemd-sysext` runs `Before=sysinit.target`.

Reply:
Good, I'll try to test again with this test case and make sure it won't have to restart :)